packages/hardhat-core/src/internal/solidity/resolver.ts

```diff
@@ -534,4 +559,14 @@ export class Resolver {
       throw error;
     }
   }
+
+  private _applyRemapping(sourceName: string): string {
+    for (const [from, to] of Object.entries(this._remappings)) {
+      if (sourceName.startsWith(from)) {
+        return sourceName.replace(from, to);
+      }
+    }
+
+    return sourceName;
+  }
```

Comment:
This is incorrect. The full algorithm applies the remapping with the longest matching prefix and, if there is a tie, the first one wins.

We should extract this to a function or something and unit-test it.